### PR TITLE
charest: add all mysql character information in parser

### LINKED ISSUE #31598: Parser support for _ujis, _CP932 literal
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

see the following testcase. While ujis is not supported as a character set, it is useful to add the parser syntax and a more useful error message to explain this.
```sql
Select CAST((LAG(_UJIS '8CA} ',144) OVER()) AS TIME);
Select LAST_DAY(LEAD( _CP932 '',241)OVER(RANGE UNBOUNDED PRECEDING));
```
### 2. What did you expect to see? (Required)
```sql
mysql> Select CAST((LAG(_UJIS '8CA} ',144) OVER()) AS TIME);
+-----------------------------------------------+
| CAST((LAG(_UJIS '8CA} ',144) OVER()) AS TIME) |
+-----------------------------------------------+
| NULL                                          |
+-----------------------------------------------+
1 row in set (0.00 sec)

mysql> Select LAST_DAY(LEAD( _CP932 '',241)OVER(RANGE UNBOUNDED PRECEDING));
+---------------------------------------------------------------+
| LAST_DAY(LEAD( _CP932 '',241)OVER(RANGE UNBOUNDED PRECEDING)) |
+---------------------------------------------------------------+
| NULL                                                          |
+---------------------------------------------------------------+
1 row in set (0.00 sec)
```
### 3. What did you see instead (Required)
```sql
mysql> Select CAST((LAG(_UJIS '8CA} ',144) OVER()) AS TIME);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 30 near "'8CA} ',144) OVER()) AS TIME)" 
mysql> Select LAST_DAY(LEAD( _CP932 '',241)OVER(RANGE UNBOUNDED PRECEDING));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 31 near "'',241)OVER(RANGE UNBOUNDED PRECEDING))" 
```
### 4. What is your TiDB version? (Required)

```sql
| Release Version: v5.5.0-alpha-60-g8c70432b0-dirty
Edition: Community
Git Commit Hash: 8c70432b000d648e9ea26cc57a7fe79d2e67d341
Git Branch: master
UTC Build Time: 2022-01-07 17:47:06
GoVersion: go1.16.9
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #31598

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Let parser support all MySQL charsets.
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: parser/lexer.go [hunk 1]
```diff
@@ -314,8 +314,8 @@ func (s *Scanner) handleIdent(lval *yySymType) int {
 	if !strings.HasPrefix(str, "_") {
 		return identifier
 	}
-	cs, err := charset.GetCharsetInfo(str[1:])
-	if err != nil {
+	cs, _ := charset.GetCharsetInfo(str[1:])
+	if cs == nil {
```
Comment (morgo) [inline parser/lexer.go:318]: The second argument here is still an error. If you don't check for it, our linter will complain.

File: parser/charset/charset.go [hunk 2]
```diff
@@ -193,33 +197,35 @@ func GetCollationByID(id int) (*Collation, error) {
...
 
 	CollationGBKBin       = "gbk_bin"
 	CollationGBKChineseCI = "gbk_chinese_ci"
+)
+
+const (
+	// CharsetASCII is a subset of UTF8.
+	CharsetASCII = "ascii"
+	// CharsetBin is used for marking binary charset.
+	CharsetBin = "binary"
```
Comment (morgo) [inline parser/charset/charset.go:219]: Can you explain the rename? It's not a huge deal, but there are some users using this as a library. Renaming `CharsetBinary` to `CharsetBin` might cause them some issues. But if `Bin` is more correct, I have no problem with it.

Comment (Defined2014) [reply parser/charset/charset.go:219]: We have `CharsetBin` and `CharsetBinary` in parser before. I found we use `CharsetBin` more, so I decided delete `CharsetBinary`.

File: parser/lexer.go (hunk 1 above)
Comment (Defined2014) [reply parser/lexer.go:318]: The linter can be passed, the previous failed was caused by ci.

Comment (morgo) [reply parser/lexer.go:318]: OK. I'll trust your judgement :-)

**Review by morgo — APPROVED** [review]

**Pull request merged**